ros2topic/ros2topic/verb/echo.py

```diff
@@ -0,0 +1,148 @@
+# Copyright 2016-2017 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from collections import OrderedDict
+import importlib
+import sys
+
+import rclpy
+from rclpy.qos import qos_profile_sensor_data
+from ros2cli.node.direct import DirectNode
+from ros2topic.api import get_topic_names_and_types
+from ros2topic.api import TopicNameCompleter
+from ros2topic.verb import VerbExtension
+import yaml
+
+
+class EchoVerb(VerbExtension):
+    """Output messages from a topic."""
+
+    def add_arguments(self, parser, cli_name):
+        arg = parser.add_argument(
+            'topic_name',
+            help="Name of the ROS topic to listen to (e.g. '/chatter')")
+        arg.completer = TopicNameCompleter(
+            include_hidden_topics_key='include_hidden_topics')
+        parser.add_argument(
+            'message_type', nargs='?',
+            help="Type of the ROS message (e.g. 'std_msgs/String')")
+        parser.add_argument(
+            '--csv', action='store_true',
+            help='Output all recursive fields separated by commas (e.g. for '
+                 'plotting)')
+
+    def main(self, *, args):
+        return main(args)
+
+
+def main(args):
+    if not args.csv:
+        register_yaml_representer()
+        callback = subscriber_cb
+    else:
+        callback = subscriber_cb_csv
+    with DirectNode(args) as node:
+        subscriber(node, args.topic_name, args.message_type, callback)
+
+
+def register_yaml_representer():
+    # Register our custom representer for YAML output
+    yaml.add_representer(OrderedDict, represent_ordereddict)
+
+
+# Custom representer for getting clean YAML output that preserves the order in
+# an OrderedDict.
+# Inspired by:
+# http://stackoverflow.com/a/16782282/7169408
+def represent_ordereddict(dumper, data):
```

Comment:
I don't see this function more than once. Can you please point to the duplicate locations.